### PR TITLE
Implement downloading WebKitTestRunner in `wpt`

### PR DESCRIPTION
**Pull request by gsnedders**
~This is currently on top of #39033, but the last commit here is \~basically correct.~

This copies over `download_url` from #39033, but is otherwise unrelated to that PR now. This suffices for `./wpt run --log-mach - --install-browser --manifest MANIFEST.json --metadata infrastructure/metadata wktr infrastructure`.

### CODE REVIEW

**Comment by gsnedders** [comment]
(sorry for the reviewer nonsense, computers are bad)

File: tools/wpt/browser.py [hunk 1]
```diff
@@ -1891,23 +1921,108 @@ class WebKitTestRunner(Browser):
...
+            platform_key = "mac-monterey-x86_64%20arm64"
+        else:
+            raise ValueError(
+                f"don't know what platform to use for macOS {system_version}"
+            )
 
-    def install_webdriver(self, dest=None, channel=None, browser_binary=None):
+        # This should match http://github.com/WebKit/WebKit/blob/main/Websites/webkit.org/wp-content/themes/webkit/build-archives.php
+        build_index = get(
+            f"https://q1tzqfy48e.execute-api.us-west-2.amazonaws.com/v3/latest/{platform_key}-release"
```
Comment (jgraham) [inline tools/wpt/browser.py:1940]: This is a terrifyingly temporary looking URL which I'd prefer if we could look up somewhere, but ofc it's up to you.

**Review by jgraham — APPROVED** [review]
Well I didn't really review it for correctness, but if it works I'm happy for the code to land.

File: tools/wpt/browser.py (hunk 1 above)
Comment (gsnedders) [reply tools/wpt/browser.py:1940]: We've changed the `v3` twice in the past six years, I'm not super concerned about the stability here. Though yeah, it doesn't seem great…

**Pull request merged**